sdk/identity/identity/review/identity.api.md

```diff
@@ -212,19 +241,28 @@ export const logger: AzureLogger;
 export class ManagedIdentityCredential implements TokenCredential {
     constructor(clientId: string, options?: TokenCredentialOptions);
     constructor(options?: TokenCredentialOptions);
-    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken>;
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
     }
 
 // @public
 export function serializeAuthenticationRecord(record: AuthenticationRecord): string;
 
+// @public
+export interface TokenCachePersistenceOptions {
+    allowUnencryptedStorage?: boolean;
+    name?: string;
+}
+
 export { TokenCredential }
 
 // @public
 export interface TokenCredentialOptions extends PipelineOptions {
     authorityHost?: string;
 }
 
+// @public
+export function useIdentityExtension<Extension extends IdentityExtension | PromiseLike<IdentityExtensionModule>>(extension: Extension): Extension extends PromiseLike<unknown> ? Promise<void> : void;
```

Comment:
so sometimes extensions are async and sometime they're not? Why not just make it always async?

Reply:
The idea here is to support writing `await useIdentityExtension(import("@azure/identity-persistence"));` as a convenience feature, but also allow just loading it in the module body, in which case we must load the extension synchronously. Most people aren't going to have top-level await.

If you pass something that looks like a `Promise`, then we return a Promise. If you pass something that is just an extension, we return nothing. The Promise input type only allows things that are shaped like modules that default-export an extension: `{ default: IdentityExtension }`.

Reply:
Why the impetus to solve the async import ourselves? Wouldn't `useIdentityExtension(await import("@azure/identity-persistence"));` work?

Reply:
Good point. The only real reason I did it this way is because I think `await useIdentityExtension(import("x"))` is quite elegant, but it's probably not worth the added complexity in the type signature.